test/service_test.cc

```diff
@@ -111,6 +113,47 @@ class ServiceTest : public TestWithParam<Envoy::Network::Address::IpVersion> {
   int grpc_server_port_{0};
 };
 
+class ServiceTestWithParameterizedConstructor : public ServiceTest {
+public:
+  void SetUp() override {
+    auto logging_context = std::make_unique<Envoy::Logger::Context>(
+        spdlog::level::info, "%L %n [%g:%#] %v", log_lock_, false);
+    service_ = std::make_unique<ServiceImpl>(std::move(logging_context));
+    grpc::ServerBuilder builder;
+    loopback_address_ = Envoy::Network::Test::getLoopbackAddressUrlString(GetParam());
+
+    builder.AddListeningPort(fmt::format("{}:0", loopback_address_),
+                             grpc::InsecureServerCredentials(), &grpc_server_port_);
+    builder.RegisterService(service_.get());
+    server_ = builder.BuildAndStart();
+    setupGrpcClient();
+    setBasicRequestOptions();
+  }
+
+private:
+  Envoy::Thread::MutexBasicLockable log_lock_;
+};
+
+INSTANTIATE_TEST_SUITE_P(IpVersions, ServiceTestWithParameterizedConstructor,
+                         ValuesIn(Envoy::TestEnvironment::getIpVersionsForTest()),
+                         Envoy::TestUtility::ipTestParamsToString);
+
+TEST_P(ServiceTestWithParameterizedConstructor,
+       ConstructorWithLoggingContextParameterCanRespondToRequests) {
+  std::unique_ptr<::grpc::ClientReaderWriter<::nighthawk::client::ExecutionRequest,
```

Comment:
Note that we are actually within namespace ::nighthawk::client, so you can just say ExecutionRequest and ExecutionResponse with no namespace declarations.  Should make this more concise.

Reply:
It actually complains about _Unknown type name 'ExecutionRequest'; did you mean 'nighthawk::client::ExecutionRequest'?_ after removing ::nighthawk::client. Maybe because I added an empty namespace wrapper?

Reply:
Talked in person with @dubious90. The error was caused by the namespace discrepancy between the code file and proto buffer. In code, our namespaces start with uppercase, while in proto buffer they start with lowercase.